File: 2. cleaning-robot/javascript/challenge.js
// Robot controller implementation

import { robotControl, Direction } from './robot-api.js';

const rc = robotControl("../resources/room-layout-1.txt");
const api = rc.robotApi;
cleanRooms(api);
rc.evaluateResult();

// Robot API
//
// api.move(); // move robot one step forward
// api.turnLeft(); // turn robot 90 degrees to the left
// api.turnRight(); // turn robot 90 degrees to the right
// api.getDirection(); // return the current direction, e.g. Direction.RIGHT
// api.isBarrierAhead(); // true -> barrier ahead
// api.getPosition(); // returns the current robot position, e.g. { x: 1, y: 2 }
// api.getPositionAhead(); // returns the position in front of the robot, doesn't check if a barrier

function cleanRooms(api) {
    // TODO add code here
}

File: 2. cleaning-robot/java/Challenge.java
import com.financial.cleaningrobot.api.RobotApi;
import com.financial.cleaningrobot.api.RobotControl;

/**
 * Robot controller implementation
 */
public class Challenge {

    /**
     * The main entry point.
     * Don't change the code, besides the file path of the room layout.
     */
    public static void main(String[] args) throws Exception {
        RobotControl rc = new RobotControl("../resources/room-layout-1.txt");
        RobotApi api = rc.getRobotApi();
        cleanRooms(api);
        rc.evaluateResult();
    }

    /**
     * Add your implementation here.
     */
    private static void cleanRooms(RobotApi api) {
        // TODO add code here
    }
}
